cmd/send_tx.go

```diff
@@ -0,0 +1,206 @@
+package cmd
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	api "code.vegaprotocol.io/protos/vega/api/v1"
+	commandspb "code.vegaprotocol.io/protos/vega/commands/v1"
+	"code.vegaprotocol.io/vegawallet/cmd/cli"
+	"code.vegaprotocol.io/vegawallet/cmd/flags"
+	"code.vegaprotocol.io/vegawallet/cmd/printer"
+	vglog "code.vegaprotocol.io/vegawallet/libs/zap"
+	"code.vegaprotocol.io/vegawallet/network"
+	"code.vegaprotocol.io/vegawallet/node"
+	"github.com/golang/protobuf/proto"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	sendTxLong = cli.LongDesc(`
+		Send a transaction to a Vega node via the gRPC API. The command can be sent to 
+		any node of a registered network or to a specific node address.
+
+		The transaction is base64-encoded.
+	`)
+
+	sendTxExample = cli.Examples(`
+		# Send a command to a registered network
+		vegawallet send tx --network NETWORK BASE64_TRANSACTION
+
+		# Send a command to a specific Vega node address
+		vegawallet send tx --node-address ADDRESS BASE64_TRANSACTION
+
+		# Send a command with a log level set to debug
+		vegawallet send tx --network NETWORK --level debug BASE64_TRANSACTION
+
+		# Send a command with a maximum of 10 retry
+		vegawallet send tx --network NETWORK --retries 10 BASE64_TRANSACTION
+	`)
+)
+
+type SendTxHandler func(io.Writer, *RootFlags, *SendTxRequest) error
+
+func NewCmdSendTx(w io.Writer, rf *RootFlags) *cobra.Command {
+	return BuildCmdSendTx(w, SendTx, rf)
+}
+
+func BuildCmdSendTx(w io.Writer, handler SendTxHandler, rf *RootFlags) *cobra.Command {
+	f := &SendTxFlags{}
+
+	cmd := &cobra.Command{
+		Use:     "command",
+		Short:   "Send a command to a Vega node",
+		Long:    sendTxLong,
+		Example: sendTxExample,
+		RunE: func(_ *cobra.Command, args []string) error {
+			if aLen := len(args); aLen == 0 {
+				return flags.ArgMustBeSpecifiedError("transaction")
+			} else if aLen > 1 {
+				return flags.TooManyArgsError("transaction")
+			}
+			f.RawTx = args[0]
+
+			req, err := f.Validate()
+			if err != nil {
+				return err
+			}
+
+			if err := handler(w, rf, req); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&f.Network,
+		"network", "n",
+		"",
+		"Network to which the command is sent",
+	)
+	cmd.Flags().StringVar(&f.NodeAddress,
+		"node-address",
+		"",
+		"Vega node address to which the command is sent",
+	)
+	cmd.Flags().StringVar(&f.LogLevel,
+		"level",
+		zapcore.InfoLevel.String(),
+		fmt.Sprintf("Set the log level: %v", SupportedLogLevels),
+	)
+	cmd.Flags().Uint64Var(&f.Retries,
+		"retries",
+		DefaultForwarderRetryCount,
+		"Number of retries when contacting the Vega node",
+	)
+
+	return cmd
+}
+
+type SendTxFlags struct {
+	Network     string
+	NodeAddress string
+	Retries     uint64
+	LogLevel    string
+	RawTx       string
+}
+
+func (f *SendTxFlags) Validate() (*SendTxRequest, error) {
+	req := &SendTxRequest{
+		Retries: f.Retries,
+	}
+
+	if len(f.LogLevel) == 0 {
```

Comment:
Minor: do we need this check if there is a default value? Same is probably also true in `send_command.go`.

Reply:
I need to verify the behavior of the default value assignment. I thought of this in case a user where set the value as `""`.

Reply:
Ok so I confirm that setting the `--level` at `""` override the default value and thus, requires presence verification.